xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/job/DeleteRequest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.refactoring.job;
+
+import org.xwiki.model.reference.DocumentReference;
+
+/**
+ * Job request for deleting a page.
+ * 
+ * @version $Id$
+ * @since 14.4RC1
+ */
+public class DeleteRequest extends EntityRequest
+{
+    /**
+     * Key of the optional property that indicates whether the document should be send to the recycle bin or removed
+     * permanently.
+     */
+    public static final String SHOULD_SKIP_RECYCLE_BIN = "shouldSkipRecycleBin";
+
+    /**
+     * @see #getNewTarget()
+     */
+    public static final String NEW_TARGET = "newTarget";
```

Comment:
You should use something more explicit, such as "new **link** target" or "new **backlink** target".